src/homework-14.1/index.ts

```diff
@@ -0,0 +1,159 @@
+enum ItemStatusEnum {
+  New,
+  InProgress,
+  Done,
+}
+
+interface ITodoItem {
+  id: string;
+  title: string;
+  content: string;
+  createdAt: Date;
+  editedAt?: Date;
+  isDone: boolean;
+  status: ItemStatusEnum;
+}
+
+class TodoItem implements ITodoItem {
+  readonly id: string = Math.random().toString(36).substring(2, 9);
+  public title: string = "";
+  public content: string = "";
+  readonly createdAt = new Date();
+  public editedAt?: Date | undefined;
+  public isDone: boolean = false;
+  public status: ItemStatusEnum = ItemStatusEnum.New;
+
+  constructor(title: string, content: string, status: ItemStatusEnum) {
+    if (!title.trim() || !content.trim()) {
+        throw new Error('Task cannot be empty!');
+      }
+  
+    this.status = status;
+    this.content = content;
+    this.title = title;
+    this.isDone = false;
+  }
+
+  public editItem(newTitle: string, newContent: string): void {
+    if (this.title.trim()) {
+      this.title = newTitle;
+    }
+
+    if (this.content.trim()) {
+      this.content = newContent;
+    }
+
+    this.editedAt = new Date();
+  }
+
+  public changeStatus(): void {
+    this.status = ItemStatusEnum.Done;
+  }
+}
+
+class ConfirmeedTodoItem extends TodoItem {
+    private isConfirmed: boolean = false;
+
+    constructor(content: string, title: string) {
+        super(title, content, ItemStatusEnum.New);
+    }
+
+    public confirm(): void {
+        this.isConfirmed = true;
+    }
+
+    public editItem(newTitle: string, newContent: string): void {
+        if (!this.isConfirmed)
+            throw new Error('Task is not confirmed!');
+        
+        super.editItem(newTitle, newContent);
+    }
+}
+
+class TodoList {
+  items: { [id: string]: TodoItem } = {};
+
+  public addNewItem(item: TodoItem): void {
+    this.items[item.id] = item;
+  }
+
+  public get numberOfTasks(): number {
+    return Object.keys(this.items).length;
+  }
+
+  public get itemInProgress(): number {
+    return Object.entries(this.items).filter(([, item]) => item.isDone === false).length;
+  }
+
+  public getItem(id: string): TodoItem | undefined {
+    return this.items[id];
+  }
+
+  public editTask(id: string, newTitle: string, newContent: string): void {
```

Comment:
I can't find two types of notes on the editing feature. Could you please double-check it?

Reply:
I have another class ConfirmedTodoItem and there is an overloaded method editItem
https://github.com/dugasova/TypeScript/blob/ae7e375a67ea3f3a0db827318b0371dbc39d3e68/src/homework-14.1/index.ts#L65